iree/tools/CMakeLists.txt

```diff
@@ -440,6 +440,8 @@ if(${IREE_BUILD_COMPILER})
       MLIRSupport
     PUBLIC
   )
+  set_target_properties(iree_tools_iree-mlir-lsp-server
+    PROPERTIES EXCLUDE_FROM_ALL ON)
```

Comment:
Aw, this doesn't interact well with the `install` configuration.

https://source.cloud.google.com/results/invocations/5c4d1b3e-1edd-4923-bbab-f4ee81faefa0/targets/iree%2Fgcp_ubuntu%2Fcmake%2Fandroid%2Farm64-v8a%2Fpresubmit/log

```
CMake Error at iree/tools/cmake_install.cmake:216 (file):
  file INSTALL cannot find
  "/tmpfs/src/github/iree/build-host/iree/tools/iree-mlir-lsp-server": No
  such file or directory.
```

we might need to add an option to `iree_cc_binary` that adds `EXCLUDE_FROM_ALL` _and_ skips this: https://github.com/google/iree/blob/26591c80cc0ee7a73b66f914d1fe909cc024445c/build_tools/cmake/iree_cc_binary.cmake#L137-L140

Reply:
I like adding `EXCLUDE_FROM_ALL` as an option to our CMake target functions. Rather than dropping the install target, we can [pass `EXCLUDE_FROM_ALL` to it as well]( https://cmake.org/cmake/help/latest/command/install.html#:~:text=the%20CMAKE_INSTALL_DEFAULT_COMPONENT_NAME%20variable.-,EXCLUDE_FROM_ALL,-New%20in%20version). This does seem like something someone would want to install though. Did we figure out why this target is so slow?